python/rmm/_lib/device_pointer.pyx

```diff
@@ -0,0 +1,13 @@
+from rmm._lib.lib cimport c_free, cudaStream_t
+
+cdef class DevicePointer:
+    def __cinit__(self, ptr, stream=0):
+        self.c_ptr = <void*><size_t>(ptr)
+        self.c_stream = <cudaStream_t><size_t>(stream)
```

Comment:
```suggestion
        self.c_stream = <cudaStream_t><uintptr_t>(stream)
```

Reply:
cudaStream_t isn't a pointer so not sure we want to use `uintptr_t`

Reply:
Sorry I confused myself. Is `size_t` appropriate here or should it be something else?

Reply:
I think `size_t` is appropriate here but I'm not sure. @harrism do you happen to know?

Reply:
I don't know this syntax, but cudaStream_t is an opaque typedef -- it's a pointer underneath though. What is this code doing?

Reply:
`stream` will be a python integer that we want to cast to a `cudaStream_t`, sounds like we should use `uintptr_t` as the intermediate then?